docs/src/capi/explanation/in-place-upgrades.md

```diff
@@ -0,0 +1,236 @@
+# In-Place Upgrades
+
+Upgrading the Kubernetes version of the machines is sometimes a necessity. 
+Rolling upgrades are the most popular way to go, but in certain situations 
+we might want or need to go with an in-place upgrade. Examples of these 
+situations are:
+
+- Resource constraints (i.e. we can not afford spawning new machines).
+- Costly and manual setup process for each machine which we can not afford.
+
+In this section we're going to see how the orchestrated in-place upgrades 
+work in {{product}} CAPI.
+
+## Annotations
+
+In CAPI, machines are considered immutable. They can not be changed once 
+they’re created. In order to change a machine, according to CAPI design 
+decisions, we must replace it with a new one. Quoting from [CAPI concepts][1]:
+
+> From the perspective of Cluster API, all Machines are immutable: once 
+they are created, they are never updated (except for labels, annotations and 
+status), only deleted.
+
+{{product}} CAPI leverages the fact that "annotations" can be changed on 
+an already created machine. As we will later see in details, we can perform 
+an in-place upgrade by changing a machine’s annotations. Changing “labels” 
+or “status” does not make much sense, as the former is mostly used as a 
+[grouping/organization mechanism][2] and the latter is [supplied and updated 
+by Kubernetes][3].
+
+That being said, the whole idea of doing an in-place upgrade might not be 
+completely aligned with the upstream cluster API design decisions as we’re 
+essentially changing a machine’s Kubernetes version, something that’s 
+supposed to be achieved by doing a rolling upgrade (replacing old ones). 
+Either way, having the ability to perform an in-place upgrade comes with 
+many benefits, so we decided to design and implement a way to enable users 
+perform these upgrades with minimum friction.
```

Comment:
This is too much information for the user. 
Let's just point to upstream docs for the curious and say something like:
While CAPI doesn't support in-place upgrades, Canonical Kubernetes CAPI does by leveraging annotations for the implementation.